develop/smart-contracts/libraries/web3-py.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Web3.py
+description: Learn how to interact with the Asset Hub chain using the Web3 python library, deploying Solidity contracts, and interacting with deployed smart contracts.
+---
+
+# Web3.py
+
+## Introduction
+
+Interacting with blockchains typically requires an interface between your application and the network. [Web3.py](https://web3py.readthedocs.io/en/stable/index.html){target=\_blank} offers this interface through a collection of libraries, facilitating seamless interaction with the nodes using HTTP or WebSocket protocols. 
+
+This guide illustrates how to utilize Web3.py for interactions with the Asset Hub chain.
+
+## Set Up the Project
+
+To start working with Web3.py, begin by initializing your project:
+
+```
+mkdir web3py-project
+cd web3py-project
+```
+
+Create and activate a virtual environment for your project:
+
+```
+python -m venv venv
+source venv/bin/activate
+```
+
+Next, install the Web3.py library:
+
+```
+pip install web3
+```
+
+## Set Up the Web3 Provider
+
+The [provider](https://web3py.readthedocs.io/en/stable/providers.html){target=\_blank} configuration is the foundation of any Web3.py application. The following example establishes a connection to the Asset Hub network. To use the example script, replace `INSERT_RPC_URL` with the appropriate value. For example, for the Westend Asset Hub TestNet, you can use this specific connection parameter:
+
+```python
+PROVIDER_RPC = 'https://westend-asset-hub-eth-rpc.polkadot.io'
+```
+
+The provider connection script should look something like this:
+
+```python title="connect_to_provider.py"
+--8<-- "code/develop/smart-contracts/libraries/web3-py/connect_to_provider.py"
+```
+
+With the Web3 provider set up, you can start querying the blockchain.
+For instance, to fetch the latest block number of the chain, you can use the following code snippet:
+
+???+ code "Complete script"
+
+    ```python title="fetch_last_block.py"
+    --8<-- "code/develop/smart-contracts/libraries/web3-py/fetch_last_block.py"
+    ```
+
+## Contract Deployment
+
+To deploy your contracts, you'll have previously compiled them and get two key files: an ABI (.json) file, which provides a JSON interface describing the contract's functions and how to interact with it, and a bytecode (.polkavm) file, which contains the low-level machine code executable on [PolkaVM](/polkadot-protocol/smart-contract-basics/polkavm-design#polkavm){target=\_blank} that represents the compiled smart contract ready for blockchain deployment.
```

Comment:
I would rephrase this paragraph a bit. Something like: Before deploying your contracts make sure you compiled them and obtained the two files ....